File: 2020/6_Custom_Customs/part_2.ts
import { readFileSync } from 'fs'
import { intersection, map, split, sum } from 'lodash'

const input = map(
  split(readFileSync('input.txt', 'utf-8'), '\n\n'),
  (group: string) => split(group, /\n/)
)

const commonLettersInGroups = map(input, (group) =>
  intersection(...map(group, (personAnswers) => [...personAnswers]))
)
const letterCountForGroups = map(commonLettersInGroups, (group) => group.length)

console.log(sum(letterCountForGroups))

File: 2022/3_Rucksack_Reorganization/part_2.rb
input =
  File
    .open('input.txt')
    .read
    .split("\n")
    .each_slice(3)
    .map { |strings| strings }

# ------------------------

dictionary = {}

for i in 0...26
  dictionary[i.times.reduce('a') { |a| a.next }] = i + 1
end
for i in 26...52
  dictionary[(i - 26).times.reduce('A') { |a| a.next }] = i + 1
end

print input
        .map { |group|
          group[0].chars.intersection(group[1].chars, group[2].chars)
        }
        .flatten
        .map { |letter| dictionary[letter] }
        .sum
